src/main/java/org/cyclonedx/util/ExtensionDeserializer.java

```diff
@@ -0,0 +1,255 @@
+import org.cyclonedx.model.vulnerability.Vulnerability1_0.Advisory;
+import org.cyclonedx.model.vulnerability.Vulnerability1_0.Cwe;
+import org.cyclonedx.model.vulnerability.Vulnerability1_0.Recommendation;
+import org.cyclonedx.model.vulnerability.Vulnerability1_0.Score;
+import org.cyclonedx.model.vulnerability.Vulnerability1_0.ScoreSource;
+import org.cyclonedx.model.vulnerability.Vulnerability1_0.Severity;
+import org.cyclonedx.model.vulnerability.Vulnerability1_0.Source;
+
+public class ExtensionDeserializer extends StdDeserializer<Extension>
+{
+  private static final String VULNERABILITIES = "vulnerabilities";
+  private static final String VULNERABILITY = "vulnerability";
+
+  public ExtensionDeserializer() {
+    this(Extension.class);
+  }
+
+  public ExtensionDeserializer(final Class vc) {
+    super(vc);
+  }
+
+  @Override
+  public Extension deserialize(
+      final JsonParser p, final DeserializationContext ctxt) throws IOException
+  {
+    if (p.currentName().equals(VULNERABILITIES)) {
+      if (p instanceof FromXmlParser) {
+        return processVulnerabilities(p);
+      }
+    }
+    return null;
+  }
+
+  private Extension processVulnerabilities(final JsonParser p) throws IOException {
+    TreeNode on =  p.readValueAsTree();
+    JsonNode vn = (JsonNode) on.get(VULNERABILITY);
+    List<ExtensibleType> extList = new ArrayList<>();
+    if (vn.isArray() && !vn.isEmpty()) {
+      for (JsonNode jn : vn) {
+        Vulnerability1_0 vuln = processVulnerability(jn);
+        extList.add(vuln);
+      }
+    } else {
+      Vulnerability1_0 vuln = processVulnerability(vn);
+      extList.add(vuln);
+    }
+    if (!extList.isEmpty()) {
+      return createAndReturnExtension(ExtensionType.VULNERABILITIES, extList);
+    }
+    return null;
+  }
+
+  private Vulnerability1_0 processVulnerability(final JsonNode vulnJson) {
+    Vulnerability1_0 vuln = new Vulnerability1_0();
+    for (Iterator<String> it = vulnJson.fieldNames(); it.hasNext(); ) {
+      String field = it.next();
+      switch (field) {
+        case "ref":
+          vuln.setRef(vulnJson.get(field).textValue());
+          break;
+        case "id":
+          vuln.setId(vulnJson.get(field).textValue());
+          break;
+        case "source":
+          vuln.setSource(processSource(vulnJson.get(field)));
+          break;
+        case "ratings":
+          vuln.setRatings(processRatings(vulnJson.get(field)));
+          break;
+        case "cwes":
+          vuln.setCwes(processCwes(vulnJson.get(field)));
+          break;
+        case "description":
+          vuln.setDescription(vulnJson.get(field).textValue());
+          break;
+        case "recommendations":
+          vuln.setRecommendations(processRecommendations(vulnJson.get(field)));
+          break;
+        case "advisories":
+          vuln.setAdvisories(processAdvisories(vulnJson.get(field)));
+          break;
+        default:
+          // Can't do anything so don't do anything
```

Comment:
I'd change this comment to something like `Unsupported field, skipped and no deserialized`

Reply:
ac4fbf8